Documentation/ExtensionArchitecture/Extbase/Reference/Validation/Validators/Index.rst

```diff
@@ -0,0 +1,648 @@
+:navigation-title: List of Validators
+
+..  include:: /Includes.rst.txt
+..  _extbase-validator:
+
+=======================================
+Built-in validators provided by Extbase
+=======================================
+
+This document lists all built-in Extbase validators,
+along with example usage for each using PHP attributes.
+
+..  seealso::
+    *   `Using validation for Extbase models and controllers <https://docs.typo3.org/permalink/t3coreapi:extbase-validation>`_
+    *   `Custom Extbase validator implementation <https://docs.typo3.org/permalink/t3coreapi:extbase-domain-validator>`_
+
+..  note::
+    All validators except the `NotEmptyValidator <https://docs.typo3.org/permalink/t3coreapi:extbase-validator-notempty>`_
+    accept empty values as valid. If empty values should not be possible,
+    combine these validators with the `NotEmptyValidator`.
+
+..  contents:: Validators in Extbase
+
+..  _extbase-validator-alphanumeric:
+
+AlphanumericValidator
+=====================
+
+Checks that a value contains **only letters and numbers** — no spaces, symbols,
+or special characters.
+
+This includes letters from **many languages**, not just A–Z. For example,
+letters from alphabets like **Hebrew, Arabic, Cyrillic**, and others are also
+allowed.
+
+This is useful for fields like usernames or codes where only plain text
+characters are allowed.
+
+If you want to allow any symbols (like `@`, `#`, `-`) or spaces, use the
+`RegularExpressionValidator <https://docs.typo3.org/permalink/t3coreapi:extbase-validator-regularexpression>`_
+instead.
+
+..  code-block:: php
+
+    #[Validate(['validator' => 'Alphanumeric'])]
+    protected string $username;
+
+..  _extbase-validator-boolean:
+
+BooleanValidator
+================
+
+Checks if a value matches a specific boolean value (`true` or `false`).
+
+This validator is useful when you want to enforce that a property explicitly
+evaluates to either `true` or `false`, such as for checkboxes in forms.
+
+By default, it accepts any boolean value unless the `is` option is
+set.
+
+Options:
+
+`is`
+    Interprets strings `'true'`, `'1'`, `'false'`, `'0'`.
+    Values of other types are converted to boolean directly.
+
+..  note::
+    Empty strings `''` and `null` always validate not matter the `is` option.
+    If you want to disallow empty strings an `null` combine this validator with
+    the `NotEmptyValidator <https://docs.typo3.org/permalink/t3coreapi:extbase-validator-notempty>`_.
```

Comment:
We either need this note everywhere except on `NotEmpty` or once at the top. Having it here only when the same applies to all of them does not make much sense.

Reply:
There is a message on top. However dealing with empty strings is expecially confusing in the Boolean Validators (empty strings are valid no matter if you set is to true or false) and the StringLength Validator (if minimum is higher then 0 empty strings still validate) therefore I added additional info boxes for those two cases.